app/users/v1/views.py

```diff
@@ -1,80 +1,77 @@
 from flask_restful import Resource
-from flask import make_response, jsonify, request
-from werkzeug.security import check_password_hash
-from .models import User
+from flask import make_response,jsonify, request
+from app.users.v1.models import User
+from app.utilities.validators import isValidEmail,isValidPassword,isValidUsername,isValidSpace,isBlank,isNumber
 
 class Registration(Resource):
     
     def post(self):
-        try:
-            data = request.get_json()
-            email = data['email']
-            password = data['password']
-            confirm_password = data['password']
-            firstname = data['firstName']
-            lastname = data['lastName']
-            phonenumber = data['phoneNumber']
-            username = data['userName']
-            registered = data['registered']
-            isAdmin = data['isadmin']
+        data = request.get_json()
+        email = data['email']
+        password = data['password']
+        firstname = data['firstname']
+        lastname = data['lastname']
+        username = data['username']
+        phonenumber = data['phonenumber']
         
-        if type(data['email']) != str or type(data['password']) != str or type(data['firstName']) or type(data['lastName']) or type(data['phoneNumber']) != int or type(data['userName']) != str:
+        if isValidEmail(email) and isValidSpace(email):
```

Comment:
Could we refactor this method? This nesting could be difficult to follow and maintain.

Reply:
I've refactored as suggested